File: src/services/alert.service.ts
import { ReceptionAlertPayload } from '../types/appointment';
import { config } from '../config';
import { logger } from '../utils/logger';

export class AlertService {
  /**
   * Envía una notificación urgente a la recepción de la clínica Dr. Zaragozá.
   * Usado para cancelaciones (liberar o reasignar hueco) o mensajes ambiguos (MANUAL_REVIEW).
   */
  public static async notifyReception(alert: ReceptionAlertPayload): Promise<void> {
    logger.warn(
      {
        type: alert.type,
        patientPhone: alert.patientPhone,
        patientName: alert.patientName || 'Desconocido',
        patientMessage: alert.patientMessage,
        reason: alert.reason,
      },
      `🚨 ALERTA A RECEPCIÓN DR. ZARAGOZÁ: [${alert.type}]`
    );

    // Si hay un webhook de recepción configurado (ej: canal de Slack, Teams, webhook de CRM)
    if (config.RECEPTION_ALERT_WEBHOOK_URL) {
      try {
        const response = await fetch(config.RECEPTION_ALERT_WEBHOOK_URL, {
          method: 'POST',
          headers: { 'Content-Type': 'application/json' },
          body: JSON.stringify({
            title: `[Alerta Clínica Dr. Zaragozá] ${alert.type}`,
            patient: alert.patientName || alert.patientPhone,
            phone: alert.patientPhone,
            message: alert.patientMessage,
            reason: alert.reason,
            timestamp: alert.timestamp,
            appointmentDetails: alert.appointment
              ? {
                  dateTime: alert.appointment.appointment_datetime,
                  specialist: alert.appointment.specialist,
                  status: alert.appointment.status,
                }
              : null,
          }),
        });

        if (!response.ok) {
          logger.error({ status: response.status }, 'Fallo al enviar webhook a recepción');
        } else {
          logger.info('Notificación enviada exitosamente al webhook de recepción');
        }
      } catch (err: any) {
        logger.error({ err: err.message }, 'Error de red enviando alerta al webhook de recepción');
      }
    }

    // Aquí se puede integrar nodemailer / Resend / Twilio SMS interno si se activa RECEPTION_ALERT_EMAIL
    if (config.RECEPTION_ALERT_EMAIL) {
      logger.info({ email: config.RECEPTION_ALERT_EMAIL }, 'Simulando despacho de email a recepción');
    }
  }
}

File: python_app/alert_service.py
import logging
import requests
from datetime import datetime
from zoneinfo import ZoneInfo
from typing import Optional
from config import RECEPTION_ALERT_WEBHOOK_URL, RECEPTION_ALERT_EMAIL
from types_models import Appointment

logger = logging.getLogger(__name__)

class AlertService:
    recent_alerts = []

    @classmethod
    def get_recent_alerts(cls):
        return cls.recent_alerts

    @classmethod
    def notify_reception(
        cls,
        alert_type: str,
        patient_phone: str,
        patient_message: str,
        reason: str,
        patient_name: Optional[str] = None,
        appointment: Optional[Appointment] = None
    ) -> None:
        alert_data = {
            "type": alert_type,
            "patient_name": patient_name or "Desconocido",
            "patient_phone": patient_phone,
            "patient_message": patient_message,
            "reason": reason,
            "timestamp": datetime.now(ZoneInfo("Europe/Madrid")).strftime("%H:%M:%S - %d/%m/%Y")
        }
        cls.recent_alerts.insert(0, alert_data)
        if len(cls.recent_alerts) > 50:
            cls.recent_alerts.pop()
        logger.warning(
            f"🚨 ALERTA A RECEPCIÓN DR. ZARAGOZÁ [{alert_type}] - Paciente: {patient_name or patient_phone} | "
            f"Mensaje: '{patient_message}' | Razón: {reason}"
        )

        if RECEPTION_ALERT_WEBHOOK_URL:
            try:
                payload = {
                    "title": f"[Clínica Dr. Zaragozá] {alert_type}",
                    "patient_name": patient_name or "Desconocido",
                    "patient_phone": patient_phone,
                    "patient_message": patient_message,
                    "reason": reason,
                    "appointment_details": {
                        "date_time": appointment.appointment_datetime if appointment else None,
                        "specialist": appointment.specialist if appointment else None,
                        "status": appointment.status.value if appointment else None
                    } if appointment else None
                }
                res = requests.post(RECEPTION_ALERT_WEBHOOK_URL, json=payload, timeout=5)
                if res.status_code >= 400:
                    logger.error(f"Fallo enviando alerta a webhook de recepción: {res.status_code}")
                else:
                    logger.info("Alerta de recepción entregada al webhook correctamente")
            except Exception as e:
                logger.error(f"Error de red contactando webhook de recepción: {e}")

        if RECEPTION_ALERT_EMAIL:
            logger.info(f"Despachando notificación por email a {RECEPTION_ALERT_EMAIL}")
